client/src/sagas/lifecycle.sagas.ts

```diff
@@ -0,0 +1,43 @@
+import { eventChannel, END, EventChannel } from 'redux-saga'
+import { call, takeEvery, put } from 'redux-saga/effects'
+
+import * as lifecycle from '@store/lifecycle.store'
+import { sleep } from './helper'
+
+function lifecycleSaga(): () => any {
+  return function* () {
+    const saveChannel: EventChannel<any> = yield call(clickListener, 'click', 'button.editor-post-publish-button')
```

Comment:
It seems this class is `editor-post-publish-button__button` for new, unsaved objects. And it would be good to support the classic editor as well, not just Gutenberg. There the button has the id `#publish`.

Reply:
For reference, this is a chunk of code I experimented with for some other feature but ultimately discarded:

```js
// ON SAVE stuff
window.setTimeout(() => {
  const gutenberg_save_button1 = document.querySelector("button.editor-post-publish-button");
  const gutenberg_save_button2 = document.querySelector("button.editor-post-publish-button__button");
  const classic_update_button = document.getElementById("publish");
  const is_classic = !!classic_update_button;

  const save_button =
    gutenberg_save_button1 || gutenberg_save_button2 || classic_update_button;

  const handler = (e) => {
    console.log("save was clicked");

    // do actual stuff here
    document.querySelector("#update_all_media_files").click();

    bubble();

    e.preventDefault();
    return;
  };

  const bubble = () => {
    save_button.removeEventListener("click", handler);
    // FIXME: instead of the timeout, only bubble when ready
    window.setTimeout(() => {
      save_button.click();
      save_button.addEventListener("click", handler);
    }, 500);
  };

  save_button.addEventListener("click", handler);
}, 1000);
```